src/systemtask/SystemTask.cpp

```diff
@@ -504,3 +504,14 @@ void SystemTask::PushMessage(System::Messages msg) {
     xQueueSend(systemTasksMsgQueue, &msg, portMAX_DELAY);
   }
 }
+
+TickType_t SystemTask::GetQueueTimeout() const {
+  // By default, the timeout on the queue is 100ms.
+  // It's extended to 4s in sleep mode, when no motion based wake up option is enabled.
+  TickType_t timeout = pdMS_TO_TICKS(100);
+  if (state == SystemTaskState::Sleeping && !settingsController.isWakeUpModeOn(Pinetime::Controllers::Settings::WakeUpMode::RaiseWrist) &&
+      !settingsController.isWakeUpModeOn(Pinetime::Controllers::Settings::WakeUpMode::Shake)) {
```

Comment:
`settingsController.GetNotificationStatus() == Controllers::Settings::Notification::Sleep` could be added as an additional clause to make the timeout longer when sleep mode is enabled (which disables the motion wake algorithms).

```suggestion
  if (state == SystemTaskState::Sleeping && ((!settingsController.isWakeUpModeOn(Pinetime::Controllers::Settings::WakeUpMode::RaiseWrist) &&
      !settingsController.isWakeUpModeOn(Pinetime::Controllers::Settings::WakeUpMode::Shake)) ||
      settingsController.GetNotificationStatus() == Controllers::Settings::Notification::Sleep)) {
```